neuralhydrology/training/basetrainer.py

```diff
@@ -205,11 +212,33 @@ def train_and_validate(self):
         Train the model for the number of epochs specified in the run configuration, and perform validation after every
         ``validate_every`` epochs. Model and optimizer state are saved after every ``save_weights_every`` epochs.
         """
+        if self._early_stopping == True:
+            if self._patience_early_stopping == 0:
+                LOGGER.info(f"Early stopping is set to true, but patience is not set. So, setting patience to default value of 5.")
+                self._patience_early_stopping = 5
+            if self._minimum_epochs_before_early_stopping == 0:
+                LOGGER.info(f"Early stopping is set to true, but minimum_epochs_before_early_stopping is not set. So, setting minimum_epochs_before_early_stopping to default value of 5.")
+                self._minimum_epochs_before_early_stopping = 5
+            
+            early_stopper = EarlyStopper(patience = self._patience_early_stopping, min_delta = 0.0001)
+
+        if self._dynamic_learning_rate == True:
+            if self._patience_dynamic_learning_rate == 0:
+                LOGGER.info(f"Dynamic learning rate is set to true, but patience is not set. So, setting patience to default value of 1.")
+                self._patience_dynamic_learning_rate = 1
+            if self._factor_dynamic_learning_rate == 0:
+                LOGGER.info(f"Dynamic learning rate is set to true, but factor is not set. So, setting factor to default value of 0.1.")
+                self._factor_dynamic_learning_rate = 0.1
+            if len(self.cfg.learning_rate.keys()) > 0:
+                LOGGER.info(f"Dynamic learning rate is set to true, so static learning rate is ignored.")  
+            scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(self.optimizer, mode='min', factor=self._factor_dynamic_learning_rate, patience=self._patience_dynamic_learning_rate)
```

Comment:
can you add a LOGGER.warning() here if `self.cfg.is_continue_training` is True, to note that the scheduler state is reset when resuming training?